### PR TITLE
Keep deleting lines when triggering deleteAllLeft on column 1

### PR DESCRIPTION
**Pull request by alfondotnet**
Related issue: https://github.com/Microsoft/vscode/issues/17143,

Let me know what you guys think, I would love to have this feature.

### CODE REVIEW

**Comment by alfonsoperez** [comment]
@rebornix any concerns with this change? thanks!

**Comment by rebornix** [comment]
@alfonsoperez I really like the idea and the code looks good. The only thing we forget to take into consideration is multi cursor.

Let's say we have below content 

image

Note the cursors are before the last character of each line. Then we press cmd+backspace

image

Everything looks pretty good. Now we have a problem with what's the expected result of another cmd+backspace. Do you think it should behave the same as a simple backspace

*backspace*

image

*cmd+backspace*
image


The command is called delete all left. While in this case, the cursor is at the beginning of the line, there is nothing on the left, so I would expect another Delete All Left will just delete the line break before the cursor. While right now, the result is correct but the result selections are not the same as a backspace. What do you think on this?

**Comment by rebornix** [comment]
BTW, if you play with Sublime Text, you can see that cmd+backspace works the same as backspace in this particular case. Our `	_getEndCursorState` needs to be smarter.

**Comment by alfonsoperez** [comment]
Good point!, thanks for the detailed review!, I will take a look

File: build/dependencies.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ function asYarnDependency(prefix, tree) {
 }
 
 function getYarnProductionDependencies(cwd) {
-	const raw = cp.execSync('yarn list --json', { cwd, encoding: 'utf8', env: { ...process.env, NODE_ENV: 'production' }, stdio: [null, null, 'ignore'] });
+	const raw = cp.execSync('yarn list --json', { cwd, encoding: 'utf8', env: Object.assign({}, process.env, { NODE_ENV: 'production' }), stdio: [null, null, 'ignore'] });
```
Comment (alfonsoperez) [inline build/dependencies.js]: my project didn't build with this spread here

File: src/vs/editor/contrib/linesOperations/linesOperations.ts [hunk 2]
```diff
@@ -408,8 +408,8 @@ export abstract class AbstractDeleteAllToBoundaryAction extends EditorAction {
 		effectiveRanges.push(rangesToDelete[rangesToDelete.length - 1]);
 
 		let endCursorState = this._getEndCursorState(primaryCursor, effectiveRanges);
+
 		let edits: IIdentifiedSingleEditOperation[] = effectiveRanges.map(range => {
-			endCursorState.push(new Selection(range.startLineNumber, range.startColumn, range.startLineNumber, range.startColumn));
```
Comment (alfonsoperez) [inline src/vs/editor/contrib/linesOperations/linesOperations.ts]: I didn't understand this line, wouldn't this produce duplicate cursors?

**Comment by alfonsoperez** [comment]
@rebornix Sorry for the delay, I have been very busy, I had some spare time to look at this, and added 2 commits, let me know what you think!

**Comment by antoniobg** [comment]
any updates on this?

**Comment by rebornix** [comment]
@alfonsoperez thanks for your contribution ❤️

**Pull request merged**